server/src/main/java/org/elasticsearch/index/IndexVersions.java

```diff
@@ -87,6 +87,7 @@ private static IndexVersion def(int id, Version luceneVersion) {
     public static final IndexVersion NEW_SPARSE_VECTOR = def(8_500_001, Version.LUCENE_9_7_0);
     public static final IndexVersion SPARSE_VECTOR_IN_FIELD_NAMES_SUPPORT = def(8_500_002, Version.LUCENE_9_7_0);
     public static final IndexVersion UPGRADE_LUCENE_9_8 = def(8_500_003, Version.LUCENE_9_8_0);
+    public static final IndexVersion ES_VERSION_8_12 = def(8_500_004, Version.LUCENE_9_8_0);
```

Comment:
For my understanding, how does this new IndexVersion get used, since I do not see it anywhere used in this PR?

Reply:
It means that `org.elasticsearch.index.IndexVersion#current` is strictly greater in `main` than in `8.11` which allows us to enforce the right version barrier between `8.11.1-SNAPSHOT` and `8.12.0-SNAPSHOT` builds. Otherwise they have the same `org.elasticsearch.index.IndexVersion#current` which makes ES think that their indices are compatible, which breaks some test assumptions.

Reply:
@kingherc, `def` adds this version into `IndexVersions#IDS`. It is used later via `LATEST_DEFINED = VERSION_IDS.lastEntry().getValue();`

Reply:
nit: can the name be more tailored to what this is guarding? As it is, others might think a new "ES_VERSION_XXX" is needed for ever minor, yet this is specifically a guard for a bug around snapshots metadata in 8.11 right?

Reply:
There is nothing preventing such thing from happening during the next release. I would add it every time we release for simplicity.

Reply:
++ I think this needs a discussion about how to handle it long-term, whether we want to preserve this behaviour (and today's tests) or not. If we want to do this every time, we need to update the release/version-bump process.

Reply:
I thought this was a different issue. I've added this topic for discussion to the core/infra agenda.

Reply:
After discussing with core/infra, I think the test issue can be assigned over to core/infra for resolution. We don't want to add new IndexVersions here, but rather slightly relax our "nothing can have semantic version checks" rules. We will follow up with a couple changes:
* Make the snapshot metadata check only occur for on-prem, and go back to semantic versions
* Add facilities for semantic version checks that will only run in serverless. This would be for things that serverless doesn't need to care about (like this too-new-index check).